test/dommock/DOMMockBuilder.ts

```diff
@@ -1115,6 +1099,125 @@ export default class DOMMockBuilder {
         ]);
       },
     };
+
+    GlobalAny.HTMLVideoElement = class MockHTMLVideoElement {
+      refSrcObject: MediaStream;
+      width: number;
+      height: number;
+
+      poster: string;
+      videoHeight: number;
+      videoWidth: number;
+      private listeners: { [type: string]: MockListener[] } = {};
+
+      private clearAttribute(): void {
+        this.videoHeight = 0;
+        this.videoWidth = 0;
+        this.width = 0;
+        this.height = 0;
+      }
+
+      set srcObject(stream: MediaStream | null) {
+        if (stream === null) {
+          this.clearAttribute();
+        }
+        this.refSrcObject = stream;
+      }
+
+      get srcObject(): MediaStream {
+        return this.refSrcObject;
+      }
+
+      addEventListener(type: string, listener: (event?: Event) => void): void {
+        if (!this.listeners.hasOwnProperty(type)) {
+          this.listeners[type] = [];
+        }
+        this.listeners[type].push(listener);
+      }
+
+      pause(): void {}
+
+      play(): void {
+        if (!!this.refSrcObject) {
+          new TimeoutScheduler(mockBehavior.videoElementStartPlayDelay).start(() => {
+            this.dispatchEvent(new Event('timeupdate'));
+          });
+
+          new TimeoutScheduler(mockBehavior.videoElementSetWidthHeightAttiributeDelay).start(() => {
+            this.videoWidth = 1280;
+            this.videoHeight = 720;
+          });
+        }
+      }
+      fired = false;
+      load(): void {
+        if (!!this.refSrcObject) {
```

Comment:
You never need `!!` inside a conditional. Remove them all.

Reply:
fixed.